tests/lib/rules/no-deprecated-slot-attribute.js

```diff
@@ -0,0 +1,297 @@
+'use strict'
+
+const RuleTester = require('eslint').RuleTester
+const rule = require('../../../lib/rules/no-deprecated-slot-attribute.js')
+
+const tester = new RuleTester({
+  parser: 'vue-eslint-parser',
+  parserOptions: {
+    ecmaVersion: 2015
+  }
+})
+
+tester.run('no-deprecated-slot-attribute', rule, {
+  valid: [
+    `<template>
+      <LinkList>
+        <a v-slot:name />
+      </LinkList>
+    </template>`,
+    `<template>
+      <LinkList>
+        <a #name />
+      </LinkList>
+    </template>`,
+    `<template>
+      <LinkList>
+        <a v-slot="{a}" />
+      </LinkList>
+    </template>`,
+    `<template>
+      <LinkList>
+        <a #default="{a}" />
+      </LinkList>
+    </template>`,
+    `<template>
+      <LinkList>
+        <a />
+      </LinkList>
+    </template>`
+  ],
+  invalid: [
+    {
+      code: `
+      <template>
+        <LinkList>
+          <a slot />
+        </LinkList>
+      </template>`,
+      output: `
+      <template>
+        <LinkList>
+          <a v-slot />
+        </LinkList>
+      </template>`,
+      errors: [
+        {
+          message: '`slot` attributes are deprecated.',
+          line: 4
+        }
+      ]
+    },
+    {
+      code: `
+      <template>
+        <LinkList>
+          <a slot="name" />
+        </LinkList>
+      </template>`,
+      output: `
+      <template>
+        <LinkList>
+          <a v-slot:name />
```

Comment:
Hmm, sorry, I had overlooked this in the previous review.

In some cases, `slot` and `v-slot` should be in different places.

- `v-slot` must be on a `<template>` element which is a direct child of a custom element.
- `v-slot` for the default slot can be on a custom element directly if there are no named slots.

So in this case, the `<a>` element must be wrapped by a `<template>` element.

In [the example of nested default slots](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0001-new-slot-syntax.md#nested-default-slots), it will be moved to the parent element.

Therefore, autofix may be tough to implement... 🤔

Reply:
Sorry for the late reply.
Thank you for pointing this out.

Sorry, I did not understand `v-slot` correctly.
I will change it so that it will only autofix if `slot` is attached to `<template>`.